loki/transform/dependency_transform.py

```diff
@@ -105,6 +105,14 @@ def transform_subroutine(self, routine, **kwargs):
             # Re-generate C-style interface header
             self.generate_interfaces(routine)
 
+        # Ensure Fortran module imports appear before 'IMPLICIT NONE'
+        intrs = [i for i in FindNodes(Intrinsic).visit(routine.spec) if i.text.lower() == 'implicit none']
+        imports = [i for i in FindNodes(Import).visit(routine.spec) if not i.c_import]
+        if len(intrs) == 1 and imports:
```

Comment:
I think this implementation will fail in cases, where interfaces appear after a variable declaration (which is allowed by the standard). There, the interface will be replaced by the import, which is still after the variable declaration. Subsequently, the implicit none will end up behind a variable declaration, which is not standard-compliant.

Reply:
Yes absolutely, I should have studied the c-style import headers a bit more carefully, this is the way to do it. Thanks a lot for pointing this out, I'll update this PR.